resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesUtilsSuite.scala

```diff
@@ -65,4 +73,49 @@ class KubernetesUtilsSuite extends SparkFunSuite {
     assert(sparkPodWithNoContainerName.pod.getSpec.getHostname == HOST)
     assert(sparkPodWithNoContainerName.container.getName == null)
   }
+
+  test("SPARK-38201: check uploadFileToHadoopCompatibleFS with different delSrc and overwrite") {
+    val uploadFileToHadoopCompatibleFSMethod =
+      PrivateMethod[Unit](Symbol("uploadFileToHadoopCompatibleFS"))
+    withTempDir { srcDir =>
+      val fileName = "test.txt"
+      val srcFile = new File(srcDir, fileName)
+      FileUtils.write(srcFile, "test", Charset.defaultCharset())
+      withTempDir { destDir =>
+        import org.apache.spark.SparkException
+        val src = new Path(srcFile.getAbsolutePath)
+        val dest = new Path(destDir.getAbsolutePath, fileName)
+        val fs = src.getFileSystem(new Configuration())
+        // Scenario 1: delSrc = false and overwrite = true, upload successful
+        KubernetesUtils
+          .invokePrivate(uploadFileToHadoopCompatibleFSMethod(src, dest, fs, false, true))
+        val firstUploadTime = fs.getFileStatus(dest).getModificationTime
+        // sleep 1s to ensure that the `ModificationTime` changes.
+        TimeUnit.SECONDS.sleep(1)
```

Comment:
`1s` is too high. Shall we minimize this delay?

Reply:
The result `fs.getFileStatus(dest).getModificationTime`  is `xxxx000` millis when `fs` is a `LocalFileSystem`, the last three digits are 0.

Reply:
Since the unit of `getModificationTime` is a millisecond , can we try `TimeUnit.MILLISECONDS.sleep(100)`?

Reply:
Oh, got it.

Reply:
In that case, we had better check the actual content of file instead of the modification time.

Reply:
Please use `FileUtils.readFileToString`.

Reply:
OK

Reply:
When it comes to the test time, Apache Spark community is very careful and has been trying to reduce the test time again and again. `Sleep` is one of the typical pattern we want to avoid.